### PR TITLE
[docs] Clarify torch.unique behavior with dim parameter

### LINKED ISSUE #166143: Feedback about torch.unique
There is the following issue on this page: https://docs.pytorch.org/docs/stable/generated/torch.unique.html

The documentation does *not* mention the important nuance when using the argument _dim_ - since the output has to have equal sizes in the dimensions not specified by _dim_ , we will get some repetitions, rather than a unique set of values, to force all the unique sets to be represented in equal length torch tensors.

Example:
```
>>> import torch
>>> x = torch.tensor([[1, 3, 2, 3], [1,2,1,2]], dtype=torch.long)
>>> torch.unique(x,dim=0)
tensor([[1, 2, 1, 2],
        [1, 3, 2, 3]])
>>> torch.unique(x,dim=1)
tensor([[1, 2, 3],
        [1, 1, 2]])
```

cc @svekars @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by vishalgoyal316**
When dim is specified, torch.unique finds unique sub-tensors (rows/columns), not unique scalar values. This means individual values may appear multiple times in the output if they exist in different sub-tensors.

Added:
- Enhanced dim parameter description with important note
- Clarified return value documentation
- Added illustrative example showing value repetition with dim=0 and dim=1

Fixes #166143

### CODE REVIEW

**Comment by vishalgoyal316** [comment]
cc: @malfet

**Review by jbschlosser — APPROVED** [review]
